packages/core-mobile/app/services/earn/EarnService.ts

```diff
@@ -254,9 +254,8 @@ class EarnService {
     })
     Logger.trace('txID', txID)
 
-    const avaxProvider = NetworkService.getProviderForNetwork(
-      avaxXPNetwork
-    ) as Avalanche.JsonRpcProvider
+    const avaxProvider =
+      ModuleManager.avalancheModule.getProvider(avaxXPNetwork)
```

Comment:
also did some cleaning up around the provider logic to make sure we utilize vm module for it